components/post_view/channel_intro_message/channel_intro_message.tsx

```diff
@@ -78,22 +112,36 @@ export default class ChannelIntroMessage extends React.PureComponent<Props> {
+                classes = 'channelIntroButton';
+                return classes;
+            } else if (this.props.theme?.type === 'Onyx' && withSvg) {
+                classes = 'channelIntroButton withSvg';
+                return classes;
+            } else if (!withSvg) {
+                classes = 'channelIntroButton';
+                return classes;
+            }
+            classes = 'channelIntroButton withSvg';
+            return classes;
+        };
+
         if (channel.type === Constants.DM_CHANNEL) {
-            return createDMIntroMessage(channel, centeredIntro, teammate, teammateName, boardComponent);
+            return createDMIntroMessage(channel, centeredIntro, this.props.isFavorite, isSetHeaderOpen, this.toggleFavorite, teammate, teammateName, boardComponent, createClasses);
         } else if (channel.type === Constants.GM_CHANNEL) {
-            return createGMIntroMessage(channel, centeredIntro, channelProfiles, currentUserId, boardComponent);
+            return createGMIntroMessage(channel, centeredIntro, channelProfiles, isInvitingPeople, isNotificationsOpen, isSetHeaderOpen, currentUserId, currentUser, this.props.isFavorite, this.toggleFavorite, isMobileView, boardComponent, createClasses);
         } else if (channel.name === Constants.DEFAULT_CHANNEL) {
-            return createDefaultIntroMessage(channel, centeredIntro, stats, usersLimit, enableUserCreation, isReadOnly, teamIsGroupConstrained, boardComponent);
-        } else if (channel.name === Constants.OFFTOPIC_CHANNEL) {
-            return createOffTopicIntroMessage(channel, centeredIntro, stats, usersLimit, boardComponent);
+            return createDefaultIntroMessage(channel, centeredIntro, stats, usersLimit, locale, creatorName, isNotificationsOpen, isSetHeaderOpen, currentUser, this.props.isFavorite, this.toggleFavorite, enableUserCreation, isReadOnly, teamIsGroupConstrained, boardComponent, createClasses);
         } else if (channel.type === Constants.OPEN_CHANNEL || channel.type === Constants.PRIVATE_CHANNEL) {
-            return createStandardIntroMessage(channel, centeredIntro, stats, usersLimit, locale, creatorName, boardComponent);
+            return createStandardIntroMessage(channel, centeredIntro, stats, usersLimit, currentUser, locale, creatorName, isInvitingPeople, isNotificationsOpen, isSetHeaderOpen, boardComponent, createClasses);
         }
         return null;
     }
 }
 
-function createGMIntroMessage(channel: Channel, centeredIntro: string, profiles: UserProfileRedux[], currentUserId: string, boardComponent?: PluginComponent) {
+function createGMIntroMessage(channel: Channel, centeredIntro: string, profiles: UserProfileRedux[], isInvitingPeople: boolean, isNotificationsOpen: boolean, isSetHeaderOpen: boolean, currentUserId: string, currentUser: UserProfileRedux, isFavorite: boolean, toggleFavorite: () => void, isMobileView: boolean, boardComponent?: PluginComponent, createClasses?: (withSvg: boolean) => string) {
```

Comment:
can we restructure this to use `options` object instead?

that way we can define a type beforehand and reduce the complexity of the function by a ton:

```
type CreatIntroMessageOptions = {
    channel: Channel;
    centeredIntro: string;
    profiles: UserProfileRedux[];
    isInvitingPeople: boolean;
    isNotificationsOpen: boolean;
    isSetHeaderOpen: boolean;
    currentUserId: string;
    currentUser: UserProfileRedux;
    isFavorite: boolean;
    toggleFavorite: () => void;
    isMobileView: boolean;
    boardComponent?: PluginComponent;
}

function createGMIntroMessage(options: CreatIntroMessageOptions) => ...
```

Reply:
I'd agree with using an object here because of the number of options we're now passing in, but I think that might be a good time to change these into components since a function that takes a single object as a parameter and returns React components is basically just a function component.

```js
type GMIntroMessageProps = {...};

function GMIntroMessage(props: GMIntroMessageProps) {...}
```

If we're already moving those into an object, the only other change would be to have `ChannelIntroMessage` render them as JSX components instead of just returning the result of a function.

Reply:
Let's just leave those in the same file for now though to minimize the size of the diff for the other reviewers. If ESLint complains about that, we can just disable that rule in this one file for now.

We can move them into their own files in a followup PR after everything else is done

Reply:
@hmhealey I had the same thought